### ISSUE TITLE
[Q] How to find run_id from a Run instance?

### ISSUE THREAD

**Issue by backpropper** [wandb/wandb#4473]
I want to extract the run_id, shown on the UI, through the API. Can you please let me know if it is possible to do so, and if so how?
Thanks!

**Comment by thanos-wandb** [comment]
Hi @backpropper thank you for your question. Are you interested to get this information during a run you've initialised with `wandb.init()` call? You can then get the ID from `wandb.run.id`, and the random assigned name as `wandb.run.name`. Would this work for you?

**Comment by backpropper** [comment]
Yes that works! Thanks!

**Comment by thanos-wandb** [comment]
Glad this worked for you @backpropper, I am closing this ticket for now and feel free to reach out again if you run into any other issue or have further questions!

**Comment by drscotthawley** [comment]
I know this is closed, but I'll write here first instead of opening a new Issue just yet: 

The above works fine for single-process runs, but it seems to return `None` when I'm running in parallel: 
```python
   print("wandb.run.id = ",wandb.run.id)
AttributeError: 'NoneType' object has no attribute 'id'
```
I'm noticing that PyTorch Lightning's `WandbLogger` has a `.version` value that is the run id on process 0, but on others, the version number is None.

**Comment by kargarisaac** [comment]
What if I want to get that info from a finished run? I only set project and name in wandb.init() in one script, and after it is done, I want to get the logged data. I use the following code and looking for a way to get the run ID in the API and not the UI:

```
# Authenticate
wandb.login(key=api_key)

# Access the run
api = wandb.Api()
run = api.run(wandb_run_path)
```

**Comment by afcruzs** [comment]
for anyone bumping into this, doing the following seems to work for a finished run:
```
 run.name = 'your_new_name'
 run.update()
```

**Comment by Chacalerks** [comment]
How can i get the run_id for a run that already finished ? it because I want to update a the config data: [https://docs.wandb.ai/guides/track/public-api-guide#update-config-for-an-existing-run](https://docs.wandb.ai/guides/track/public-api-guide#update-config-for-an-existing-run).

I'm a beginner using wandb

**Comment by Chacalerks** [comment]
How can i get the run_id for a run that already finished ? it because I want to update a the config data: https://docs.wandb.ai/guides/track/public-api-guide#update-config-for-an-existing-run.

I'm a beginner using wandb, @kargarisaac said using the `wandb_run_path` but what its exactly this string ?

**Comment by Chacalerks** [comment]
i found this in the documentation, it works [https://docs.wandb.ai/ref/python/public-api/api#runs](https://docs.wandb.ai/ref/python/public-api/api#runs) 

```python
import wandb

api = wandb.Api()
runs = api.runs(path="<my_entity>/image-my_project")
for i in runs:
  print("run name = ",i.name," id: ", i.id)
```
